spyder/plugins/completion/languageserver/client.py

```diff
@@ -288,74 +289,81 @@ def patched_cleanup():
         logger.debug('LSP {} client started!'.format(self.language))
 
     def stop(self):
+        """Stop transport and server."""
         logger.info('Stopping {} client...'.format(self.language))
         if self.notifier is not None:
             self.notifier.activated.disconnect(self.on_msg_received)
             self.notifier.setEnabled(False)
             self.notifier = None
-        if self.transport_client is not None:
-            self.transport_client.kill()
+        if self.transport is not None:
+            self.transport.kill()
         self.context.destroy()
-        if self.lsp_server is not None:
-            self.lsp_server.kill()
+        if self.server is not None:
+            self.server.kill()
 
     def is_transport_alive(self):
         """Detect if transport layer is alive."""
-        alive = True
-        if self.transport_client is not None:
-            if self.transport_client.poll() is not None:
-                alive = False
-
-        return alive
+        state = self.transport.state()
+        if state == QProcess.Starting:
+            return True
+        else:
+            return state != QProcess.NotRunning
```

Comment:
Why not make this just

`return  state != QProcess.NotRunning`

If it is Starting or Running it will be True, if it is Not Running it will be False

Reply:
Yoiu're right. Done.